From 06fd00f65cc3dba1718cc515fcd41e608b5400ef Mon Sep 17 00:00:00 2001
From: Nicolas LOUBOUTIN <nicolas.louboutin@st.com>
Date: Thu, 29 Feb 2024 15:14:17 +0100
Subject: [PATCH 2/2] Workaround: force GPU composition also if there is only
 one layer

Some layer composition seems not compatible with the HW capabilities.
To avoid issue on the display, we forced to the GPU to adapt the image format to the display.

Change-Id: Ia01c6edb8f4a7fdf0c90af42ef85d8f311f9db53
Signed-off-by: Nicolas LOUBOUTIN <nicolas.louboutin@st.com>
---
 backend/Backend.cpp | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/backend/Backend.cpp b/backend/Backend.cpp
index 3ca6e92..e5a3315 100644
--- a/backend/Backend.cpp
+++ b/backend/Backend.cpp
@@ -137,6 +137,12 @@ std::tuple<int, int> Backend::GetExtraClientRange(
   if (avail_planes < display->layers().size())
     avail_planes--;
 
+  /* By default, if only one layer available, consider that it has not a compatible format
+   * TODO: Check that the validated layer has a compatible format
+   */
+  if (display->layers().size() == 1)
+    avail_planes--;
+
   const int extra_client = int(layers.size() - client_size) - int(avail_planes);
 
   if (extra_client > 0) {
-- 
2.34.1

